arrows/vxl/pixel_feature_extractor.cxx

```diff
@@ -197,12 +197,31 @@ template < typename out_t, typename in_t >
 vil_image_view< out_t >
 clamping_cast( vil_image_view< in_t > input_image )
 {
+  // Safely compute the min and max given any type of inputs
+  constexpr auto casted_in_min =
+    static_cast< double >( std::numeric_limits< in_t >::min() );
+  constexpr auto casted_out_min =
+    static_cast< double >( std::numeric_limits< out_t >::min() );
+
+  constexpr auto casted_in_max =
+    static_cast< double >( std::numeric_limits< in_t >::max() );
+  constexpr auto casted_out_max =
+    static_cast< double >( std::numeric_limits< out_t >::max() );
+
   constexpr auto min_output_value =
-    static_cast< in_t >( std::numeric_limits< out_t >::min() );
+    ( casted_in_min > casted_out_min
+      ? std::numeric_limits< in_t >::min()
+      : static_cast< in_t >( std::numeric_limits< out_t >::min() ) );
+
   constexpr auto max_output_value =
-    static_cast< in_t >( std::numeric_limits< out_t >::max() );
+    ( casted_in_max < casted_out_max
+      ? std::numeric_limits< in_t >::max()
+      : static_cast< in_t >( std::numeric_limits< out_t >::max() ) );
 
-  vil_clamp( input_image, input_image, min_output_value, max_output_value );
+  if ( casted_in_min < casted_out_min || casted_in_max > casted_out_max )
```

Comment:
```suggestion
  if( casted_in_min < casted_out_min || casted_in_max > casted_out_max )
```

Reply:
Also, as a minor aside, `min_output_value` and `max_output_value` could be moved inside the `if`. However as they are `constexpr`, it isn't very important.